### PR TITLE
WIP: JDK17 & Apple M1 Silicon compatibility

### PR DESCRIPTION
**Pull request by cchepelov**

### CODE REVIEW

File: basestar-event/src/test/java/io/basestar/event/TestEventSerialization.java [hunk 1]
```diff
@@ -67,6 +98,53 @@ void testGzipBson() {
         assertNotNull(event.getAfter());
         final byte[] outputBytes = EventSerialization.gzipBson().serialize(event);
         final String output = BaseEncoding.base64().encode(outputBytes);
-        assertEquals(input, output);
+
```
Comment (cchepelov) [inline basestar-event/src/test/java/io/basestar/event/TestEventSerialization.java:101]: I clearly went the long route here. 
Upon re-reading, it seems it has less to do with Apple M1 than with JDK17 using a more recent version of the gzip library, which drops the out of date "OS" header field. 

@arcneon, do you want me to simplify this, keeping only the "nuke 10th byte" logic?

File: basestar-database-server/src/test/java/io/basestar/database/TestDatabaseServer.java [hunk 2]
```diff
@@ -274,11 +279,15 @@ void updateSimple() throws Exception {
...
 
-        verify(emitter, times(1))
-                .emit(ObjectUpdatedEvent.of(SIMPLE, id, 1L, create, update));
+        log.info(" VERIFICATION PHASE OF UPDATE SIMPLE") ;
+
+        final InOrder inOrder = Mockito.inOrder(emitter);
+        inOrder.verify(emitter).emit(ObjectUpdatedEvent.of(SIMPLE, id, 1L, create, update));
+            //dont emmit events for identical values updates
+        // inOrder.verify(emitter).emit(any(ObjectUpdatedEvent.class));
+        inOrder.verifyNoMoreInteractions();
```
Comment (cchepelov) [inline basestar-database-server/src/test/java/io/basestar/database/TestDatabaseServer.java:288]: @arcneon help needed. I'm concerned I may have either let slipped a subtle change of main code behaviour (bad!!) under the guise of a subtle change of Mockito semantics. 
Please double check.

In any case I probably ought to clean this up once we're satisfied either way.

File: basestar-database-server/src/test/java/io/basestar/database/TestDatabaseServer.java [hunk 3]
```diff
@@ -131,12 +132,16 @@ void setUp() throws Exception {
...
+            log.info("Emitting {}", inv.getArgument(0, Event.class));
             return CompletableFuture.completedFuture(null);
         });
         when(emitter.emit(any(Collection.class))).then(inv -> {
             final Emitter emitter = (Emitter) inv.getMock();
-            inv.getArgumentAt(0, Collection.class).forEach(event -> emitter.emit((Event) event));
+            final Collection<Event> events = inv.getArgument(0, Collection.class);
+            log.info("Got collection of events with size: {}", events.size());
+            events.forEach(emitter::emit);
+            log.info("..Done emitting collection of events with size: {}", events.size());
```
Comment (cchepelov) [inline basestar-database-server/src/test/java/io/basestar/database/TestDatabaseServer.java:143]: (extra logging to be dropped once behaviour around line 286 is confirmed correct)

File: basestar-event/src/test/java/io/basestar/event/TestEventSerialization.java (hunk 1 above)
Comment (arcneon) [reply basestar-event/src/test/java/io/basestar/event/TestEventSerialization.java:101]: we shouldn't really be testing this TBH, it's only important that all instances in the same cluster can deserialize their own messages, not important that they work between JVMs.

Comment (cchepelov) [reply basestar-event/src/test/java/io/basestar/event/TestEventSerialization.java:101]: so according to this the test should actually be "serialize a message / deserialize it on the way out"

(I think I disagree though; we could have a need for being able to upgrade the JDK of components of a cluster without having first to drain all SQS queues.)

File: basestar-database-server/src/test/java/io/basestar/database/TestDatabaseServer.java (hunk 2 above)
Comment (arcneon) [reply basestar-database-server/src/test/java/io/basestar/database/TestDatabaseServer.java:288]: I'm not 100% sure what this comment/change refers to, when i run with jdk 11 i get a test pass with the original assert:

`
verify(emitter, times(1))
                .emit(ObjectUpdatedEvent.of(SIMPLE, id, 1L, create, update));
`
                
Were/are you getting an error with this assert? Can you add the error and some info about the build environment?

**Review by arcneon — APPROVED** [review]

**Pull request merged**